### PR TITLE
feature/binding-mode

### PR DESCRIPTION
**Pull request by fkleuver**
refactor(binding-mode): deprecate oneWay and addturn bindingMode intof lags
refactor(binding): change things in the runtime accordingly, and apply a few optimizations to binding.ts (both related and unrelated to binding.ts)

fix(binding/resources): update bindingBehaviors with flags and new bindingMode

test(binding/resources): add basic unit tests as a starting point
fix(signals): correct argument count

### CODE REVIEW

File: src/runtime/binding/binding.ts [hunk 1]
```diff
@@ -24,6 +24,12 @@ export interface IBinding extends IBindScope {
 
 export type IBindingTarget = any; // Node | CSSStyleDeclaration | IObservable;
 
+// BindingMode is not a const enum (and therefore not inlined), so assigning them to a variable to save a member accessor is a minor perf tweak
+const { oneTime, toView, fromView } = BindingMode;
```
Comment (EisenbergEffect) [inline src/runtime/binding/binding.ts]: Nice.

File: src/runtime/binding/binding-mode.ts [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+* Furthermore, the "default" mode would be for simple ".bind" expressions to make it explicit for our logic that the default is being used.
+* This essentially adds extra information which binding could use to do smarter things and allows bindingBehaviors that add a mode instead of simply overwriting it
+*/
 export enum BindingMode {
-  oneTime = 0,
-  toView = 1,
-  oneWay = 1,
-  twoWay = 2,
-  fromView = 3
+  oneTime  = 0b0001,
```
Comment (EisenbergEffect) [inline src/runtime/binding/binding-mode.ts]: Love this approach! Thanks for the great commenting here too.

File: src/runtime/binding/binding.ts [hunk 3]
```diff
@@ -87,42 +98,37 @@ export class Binding implements IBinding {
...
-    if (!this.targetObserver) {
-      let method: 'getObserver' | 'getAccessor' = mode === BindingMode.twoWay || mode === BindingMode.fromView ? 'getObserver' : 'getAccessor';
-      this.targetObserver = <any>this.observerLocator[method](this.target, this.targetProperty);
+    if (targetObserver.bind) {
+      targetObserver.bind(flags);
     }
 
-    if ('bind' in this.targetObserver) {
-      this.targetObserver.bind(flags);
+    if (mode & toViewOrOneTime) {
```
Comment (EisenbergEffect) [inline src/runtime/binding/binding.ts]: You've definitely sold me on the flags approach.

**Comment by fkleuver** [comment]
Appreciate the feedback!

**Pull request merged**